core/src/main/java/com/sequenceiq/cloudbreak/init/clustertemplate/DefaultClusterTemplateCache.java

```diff
@@ -93,29 +94,41 @@ private void convertToClusterTemplate(String templateAsString) throws IOExceptio
         if (defaultClusterTemplates.get(clusterTemplateRequest.getName()) != null) {
             LOGGER.warn("Default cluster template exists and it will be override: {}", clusterTemplateRequest.getName());
         }
-        defaultClusterTemplates.put(clusterTemplateRequest.getName(), clusterTemplateRequest);
+        defaultClusterTemplates.put(clusterTemplateRequest.getName(), Base64.getEncoder().encodeToString(templateAsString.getBytes()));
     }
 
-    public Map<String, DefaultClusterTemplateV4Request> defaultClusterTemplateRequests() {
-        Map<String, DefaultClusterTemplateV4Request> ret = new HashMap<>();
-        defaultClusterTemplates.forEach((key, value) -> ret.put(key, SerializationUtils.clone(value)));
-        return ret;
+    public Map<String, String> defaultClusterTemplateRequests() {
+        return defaultClusterTemplates;
     }
 
     public Map<String, ClusterTemplate> defaultClusterTemplates() {
         Map<String, ClusterTemplate> defaultTemplates = new HashMap<>();
         defaultClusterTemplateRequests().forEach((key, value) -> {
-            ClusterTemplate clusterTemplate = converterUtil.convert(value, ClusterTemplate.class);
+            String defaultTemplateJson = new String(Base64.getDecoder().decode(value));
+            DefaultClusterTemplateV4Request defaultClusterTemplate = getDefaultClusterTemplate(defaultTemplateJson);
+            ClusterTemplate clusterTemplate = converterUtil.convert(defaultClusterTemplate, ClusterTemplate.class);
             defaultTemplates.put(key, clusterTemplate);
         });
         return defaultTemplates;
     }
 
+    private DefaultClusterTemplateV4Request getDefaultClusterTemplate(String defaultTemplateJson) {
+        try {
+            return JsonUtil.readValue(defaultTemplateJson, DefaultClusterTemplateV4Request.class);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Default cluster template could not be deserilaize: " + e.getMessage(), e);
```

Comment:
Typo: Change `deserilaize` to `deserialized`.

Reply:
I think this error message is populated to the ui so can we maybe say "Default cluster template could not be added."